src/middlewared/middlewared/alert/source/smartd.py

```diff
@@ -15,9 +15,10 @@ def check_sync(self):
             if self.middleware.call_sync("system.vm"):
                 return
 
-            if not self.middleware.call_sync("system.is_freenas"):
-                if self.middleware.call_sync("failover.status") != "MASTER":
-                    return
+            is_ha = self.middleware.call_sync('failover.licensed')
+            is_master = self.middleware.call_sync('failover.status') == 'MASTER'
+            if is_ha and not is_master:
```

Comment:
The problem with this change is that now we are making two calls instead of one, if its not HA licensed